### PR TITLE
layers: Validation for VK_EXT_vertex_input_dynamic_state

### PR DESCRIPTION
**Pull request by pdaniell-nv**

### CODE REVIEW

**Comment by lunarpapillo** [comment]
I happened to be monitoring this build.  One test failed on both **Windows NVIDIA Release32** and **Windows NVIDIA Debug64** builds:
```
[ RUN      ] VkLayerTest.ValidateExtendedDynamicStateEnabled
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(185): error: Failed
Did not receive expected error 'VUID-vkCmdBindVertexBuffers2EXT-pStrides-03361'
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(185): error: Failed
Did not receive expected error 'VUID-vkCmdBindVertexBuffers2EXT-pStrides-03361'
Unexpected: Validation Error: [ VUID-vkCmdBindVertexBuffers2EXT-pStrides-03363 ] Object 0: handle = 0x9f1d054, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x880879b9 | The pStrides[0] (0) parameter in the last call to vkCmdBindVertexBuffers2EXT is less than the extent of the binding for attribute 0 (4). The Vulkan spec states: If pStrides is not NULL each element of pStrides must be greater than or equal to the maximum extent of of all vertex input attributes fetched from the corresponding binding, where the extent is calculated as the VkVertexInputAttributeDescription::offset plus VkVertexInputAttributeDescription::format size (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkCmdBindVertexBuffers2EXT-pStrides-03363)
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(195): error: Failed
Received unexpected error(s).
[  FAILED  ] VkLayerTest.ValidateExtendedDynamicStateEnabled (818 ms)
```
All tests passed on **Linux**, **Android**, and **Windows AMD** configurations.

File: layers/parameter_validation_utils.cpp [hunk 1]
```diff
@@ -6835,3 +6853,166 @@ bool StatelessValidation::manual_PreCallValidateCreatePrivateDataSlotEXT(VkDevic
...
+            device, "VUID-vkCmdSetVertexInputEXT-vertexAttributeDescriptionCount-04792",
+            "vkCmdSetVertexInputEXT(): vertexAttributeDescriptionCount is greater than the maxVertexInputAttributes limit");
+    }
+
+    // VUID-vkCmdSetVertexInputEXT-binding-04793
+    for (uint32_t attribute = 0; attribute < vertexAttributeDescriptionCount; ++attribute) {
+        bool binding_found = false;
+        for (uint32_t binding = 0; binding < vertexBindingDescriptionCount; ++binding) {
+            if (pVertexAttributeDescriptions[attribute].binding == pVertexBindingDescriptions[binding].binding) {
+                binding_found = true;
```
Comment (ncesario-lunarg) [inline layers/parameter_validation_utils.cpp:6892]: `break;`?

**Review by ncesario-lunarg — COMMENTED** [review]
A couple questions on my part, but LGTM! Will tests for these added VUIDs be added as part of this PR?

File: layers/parameter_validation_utils.cpp (hunk 1 above)
Comment (pdaniell-nv) [reply layers/parameter_validation_utils.cpp:6892]: Yeah, nice catch. Fixed.

**Comment by pdaniell-nv** [comment]
@ncesario-lunarg, thanks for your feedback. I'll add a commit for the tests shortly and remove the draft status.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -14620,19 +14628,6 @@ bool CoreChecks::PreCallValidateCmdBindVertexBuffers2EXT(VkCommandBuffer command
...
-    if (vibs && (pStrides == nullptr)) {
-        skip |= LogError(commandBuffer, "VUID-vkCmdBindVertexBuffers2EXT-pStrides-03361",
-                         "vkCmdBindVertexBuffers2EXT(): pStrides is NULL but the bound pipeline was created with "
-                         "VK_DYNAMIC_STATE_VERTEX_INPUT_BINDING_STRIDE_EXT dynamic state enabled.");
-    } else if (!vibs && (pStrides != nullptr)) {
-        skip |= LogError(commandBuffer, "VUID-vkCmdBindVertexBuffers2EXT-pStrides-03361",
-                         "vkCmdBindVertexBuffers2EXT(): pStrides is non-NULL, but the bound pipeline was not created with "
-                         "VK_DYNAMIC_STATE_VERTEX_INPUT_BINDING_STRIDE_EXT dynamic state enabled.");
-    }
-
```
[the remark this answers is not in the document]
Comment (ncesario-lunarg) [reply layers/core_validation.cpp:14747]: Thanks for the explanation, much appreciated!

**Review by ncesario-lunarg — APPROVED** [review]
Couple minor nits in the tests, but LGTM!

**Comment by pdaniell-nv** [comment]
I've renamed the branch and moved the PR here: https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2726

**Pull request closed without merging**